mavenBuild does not fail on Unit Test Failures

Hi everyone!

Running my pipelines, I noticed that `mavenExecute` ignores failures of the build step and instead continues to create build information and publish the artifact:

```
2022-03-10T14:01:33.2892194Z info  mavenBuild - [INFO] ------------------------------------------------------------------------
2022-03-10T14:01:33.2892838Z info  mavenBuild - [INFO] BUILD FAILURE
2022-03-10T14:01:33.2893531Z info  mavenBuild - [INFO] ------------------------------------------------------------------------
2022-03-10T14:01:33.2912724Z info  mavenBuild - [INFO] Total time:  03:29 min
2022-03-10T14:01:33.2913347Z info  mavenBuild - [INFO] Finished at: 2022-03-10T14:01:33Z
2022-03-10T14:01:33.2914017Z info  mavenBuild - [INFO] ------------------------------------------------------------------------
2022-03-10T14:01:33.2926795Z error mavenBuild - [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project xxx: There are test failures.
2022-03-10T14:01:33.2927846Z error mavenBuild - [ERROR] 
2022-03-10T14:01:33.2928508Z error mavenBuild - [ERROR] Please refer to /home/vsts/work/1/s/target/surefire-reports for the individual test results.
2022-03-10T14:01:33.2929322Z error mavenBuild - [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2022-03-10T14:01:33.2929921Z error mavenBuild - [ERROR] -> [Help 1]
2022-03-10T14:01:33.2930348Z error mavenBuild - [ERROR] 
2022-03-10T14:01:33.2930939Z error mavenBuild - [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2022-03-10T14:01:33.2931627Z error mavenBuild - [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2022-03-10T14:01:33.2932142Z error mavenBuild - [ERROR] 
2022-03-10T14:01:33.2932789Z error mavenBuild - [ERROR] For more information about the errors and possible solutions, please read the following articles:
2022-03-10T14:01:33.2933570Z error mavenBuild - [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
2022-03-10T14:01:33.3106213Z debug mavenBuild - creating build settings information...
2022-03-10T14:01:33.3107065Z info  mavenBuild - Printing stepName mavenBuild
2022-03-10T14:01:33.3203415Z info  mavenBuild - build settings information successfully created with '{"mavenBuild":[{"publish":true,"createBOM":true,"logSuccessfulMavenTransfers":true,"dockerImage":"maven:3.8-jdk-11"}]}
2022-03-10T14:01:33.3204243Z info  mavenBuild - publish detected, running mvn deploy
```

I believe, this issue comes from here: https://github.com/SAP/jenkins-library/blob/master/cmd/mavenBuild.go#L86

```go
	_, err := maven.Execute(&mavenOptions, utils)

	log.Entry().Debugf("creating build settings information...")
	stepName := "mavenBuild"
	dockerImage, err := getDockerImageValue(stepName)
	if err != nil {
		return err
	}
```

Maven is executed, but the error result is never used. I believe, the code is missing an additional

```java
	if err != nil {
		return err
	}
```

This must have been the case for quite some time and is particularly unhelpful if you use `mavenBuild` as PR voter, which as a consequence never fails. I also think this might not be what you want in normal pipeline runs.

Also, it is not in line with `npmExecuteScripts`, which indeed does fail if scripts executed during the `prepare` lifecycle (like linters) fail.

Please consider changing the behavior to fail the step on test failures.

Kind regards
Lukas

--- a/cmd/mavenBuild.go
+++ b/cmd/mavenBuild.go
@@ -85,6 +85,10 @@ func runMavenBuild(config *mavenBuildOptions, telemetryData *telemetry.CustomDat
 
 	_, err := maven.Execute(&mavenOptions, utils)
 
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute maven build for goal(s) '%v'", goals)
+	}
+
 	log.Entry().Debugf("creating build settings information...")
 	stepName := "mavenBuild"
 	dockerImage, err := getDockerImageValue(stepName)

